### PR TITLE
TS polishing pass: More depth sorting and starting units

### PR DESCRIPTION
**Pull request by reaperrr**
- Fixes depth sorting of editor actors (spawnpoint, waypoint, camera)
- Fixes depth sorting and offsets of several civilian structures
- Increases number of starting units for medium by 1 and for heavy by 2

### CODE REVIEW

**Review by Mailaender — APPROVED** [review]

**Comment by LipkeGu** [comment]
closes #12119

**Comment by MustaphaTR** [comment]
I'm not sure if i'm happy with the starting units changes in this PR, i think it should be like following:

GDI Light: e1, e1, e2, e2, jumpjet, smech
GDI Heavy: e1, e1, e1, e2, e2, jumpjet, smech, smech, mmch
Nod Light: e1, e1, e3, e3, cyborg, bggy
Nod Heavy: e1, e1, e1, e3, e3, cyborg, bggy, bike, ttnk

**Comment by reaperrr** [comment]
Updated.

Heavy starting forces now match what @MustaphaTR suggested, Medium now match what he suggested for Light, and on Light the e2 and e3 were replaced with a single Wolverine/Buggy respectively (Grenadiers are superior to Bazookas in direct combat, so these were quite unbalanced before).

File: mods/ts/sequences/civilian.yaml [hunk 1]
```diff
@@ -368,17 +389,22 @@ capyr03:
 ^city2x3:
 	Inherits: ^cabase
 	Defaults:
-		Offset: 12, -30, 30
+		Offset: 12, -30, 36
```
Comment (pchote) [inline mods/ts/sequences/civilian.yaml]: Why are you setting a larger z offset than y offset in these cases?

Comment (reaperrr) [inline mods/ts/sequences/civilian.yaml]: The bottom-right edges were cut off (invisible) with the previous values. If there's a better way to fix that, I'm all ears/eyes.

Comment (pchote) [inline mods/ts/sequences/civilian.yaml]: The depth sprite is probably not centered relative to the artwork, try moving that to the right instead.  If ^ is the case then adding a uniform offset to Z may indeed fix the right hand side, but makes the imbalance on the left even larger, causing potential clipping problems there instead.

**Comment by reaperrr** [comment]
Updated, used `DepthSpriteOffset` to further improve temperate civ building offsets.

**Review by pchote — CHANGES_REQUESTED** [review]
city17 is missing a couple of px off the lamp posts on the side, and gakodk could do with a fix too.  Just in case you're doing this the hard way, you can use the map editor to quickly look for cut off pixels (compare the placement preview with the placed actor).

You should also try using 4d2aebe2b822de5afaa128ace12c91fbd922fab2 (ingame, won't work in the editor) to view the relative positions of the depth mask to the actor.  The bright stripe should be aligned with the front corner of the actor, which you'll see isn't the case for a few of civ structures.

**Comment by reaperrr** [comment]
> city17 is missing a couple of px off the lamp posts on the side

Are you sure? I couldn't see a difference, and the depth sprite position looks right to me.

Anyway, improved the 4x2 buildings and added fixes for the Kodiak, Montauk, Hospital, deployed ICBM and the 5 larger `aban*` snow buildings.

**Comment by reaperrr** [comment]
@pchote I gave city17 another good, hard look, but I really can't spot a single pixel difference between the preview and placed building. If it was really that building, I'll need a screenshot, otherwise I'll assume it was either a different building or I already fixed it by accident somehow.

I did spot some grave z-sorting issues with deployed sensor arrays, though, so I'll add a fix for that.

**Comment by reaperrr** [comment]
Updated, added deployed sensor array z-sorting fix.

**Comment by pchote** [comment]
> or I already fixed it by accident somehow.

I can easily reproduce it on your old branch (which is still available behind 4d2aebe), but not on the current version.

**Review by pchote — APPROVED** [review]

**Review by pchote — APPROVED** [review]
The depth changes look good enough to merge now. :+1:

A future PR might want to add the iso depth sprite to billboards and the dam.  Some of the civ structures with dirt and junk around them could also use some tweaks, but I expect the behaviour here to match the original game, so I won't ask for that here.

**Comment by pchote** [comment]
I've attached a map with all the editor-placeable actors, which I used with a rebased  4d2aebe for testing.

[depthtest.zip](https://github.com/OpenRA/OpenRA/files/546445/depthtest.zip)

**Pull request merged**